src/index.js

```diff
@@ -18,11 +20,13 @@ ReactDOM.render(
         <Route path="about" component={About} />
         <Route path="timers" component={TimersPage} />
         <Route path="admin" component={AdminPage} />
-        <Route path="invoice" component={InvoicePage} />
+        <Route path="invoice" component={NewInvoicePage} />
         <Route path="invoices" component={InvoicesPage} />
+        <Route path="invoices/:invoiceId" component={InvoicePage} 
+          onEnter={(nextState) => {store.dispatch(getInvoice(nextState.params.invoiceId));}} />
```

Comment:
@jysandy The onEnter callback triggers the backend call to fetch Invoice detail. We can't drop that without another way to do it